source/Halibut.Tests/BackwardsCompatibility/TestOldClientWithNewService.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.BackwardsCompatibility.Util;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.BackwardsCompatibility;
+using Halibut.Tests.TestServices;
+using NUnit.Framework;
+
+namespace Halibut.Tests.BackwardsCompatibility
+{
+    public class TestOldClientWithNewService
+    {
+        [Test]
+        public async Task Listening()
+        {
+            new SerilogLoggerBuilder().Build().Information("Hello");
+            var echoService = new CallRecordingEchoServiceDecorator(new EchoService());
+            using (var clientAndService = await PreviousClientVersionAndServiceBuilder.WithListeningService().WithClientVersion(PreviousServiceVersions.v5_0_429)
```

Comment:
`WithClientVersion(PreviousServiceVersions.v5_0_429)` is a little confusing - probably want to rename `PreviousServiceVersions` to something more generic

Reply:
good idea, renamed to `PreviousVersions`